subtests/docker_cli/systemd_in_container/test.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# Test systemd in containers
+#
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 IMAGE-NAME"
+    exit 1
+fi
+image="$1"
+timeout=10
+
+# Exit status. Default 0 (success), but set to 1 on any subtest failure
+rc=0
+
+# Check that when container is run, its output contains the expected
+# string(s). With systemd, the container might be running bug systemd
+# freezing execution so we cannot really expect docker run to fail.
+check_output() {
+    local container="$1"; shift
+    local name="$1"; shift
+
+    local lrc=0
+    for expect in "$@"; do
+        echo "Checking for $expect"
+        i=0
+	while [ $i -lt $timeout ] ; do
+            if docker logs "$container" | grep -q "$expect" ; then
+                echo "PASS $name = $expect"
+                break
+	    else
+	        if [ $i -lt $timeout ] ; then
+                    echo "Did not find the output in attempt $i, will retry"
+	        fi
+	    fi
```

Comment:
Missing `i=$(expr $i + 1)`

I would also suggest adding `sleep 1`, because that would make it a real timeout (as it is, it's just a retry counter and it's possible for all ten iterations to finish before systemd has started.)